setup.py

```diff
@@ -25,6 +25,7 @@
     # Disallowing 2.6.0 can be removed when this is fixed https://github.com/pydantic/pydantic/issues/8705
     "pydantic>=1.10,<3,!=2.6.0",  # could be both V1 and V2
     "docker",
+    "pyyaml",
```

Comment:
We can add `pyyaml` to test dependencies. But for install dependency let's move this to an optional dependency like those below.

Reply:
actually I dont see any of test dependency or optional dependency under setup.py
should I put it under test from extra_require?

Reply:
You can add it like here: https://github.com/microsoft/autogen/blob/82903f5f89a4f514b267911e56c3032b339f2581/setup.py#L60-L70